core/src/oracle_state.rs

```diff
@@ -529,8 +531,8 @@ pub fn register_and_save_scans() -> std::result::Result<(), Error> {
     let datapoint_contract_address =
         OracleContract::checked_load(&config.oracle_box_wrapper_inputs.contract_inputs)?
             .ergo_tree();
-    // If scanIDs.json exists, skip registering scans & saving generated ids
-    if !Path::new("scanIDs.json").exists() {
+
+    let register_scans = || -> std::result::Result<(), Error> {
```

Comment:
Extracting this closure into a function would improve `register_and_save_scans()` overall readability.

Reply:
Sure, it's done.